Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm

```diff
@@ -241,9 +365,62 @@ - (id)init:(std::shared_ptr<Babylon::Plugins::Camera::Impl::ImplData>)implData;
     void Camera::Impl::UpdateCameraTexture(bgfx::TextureHandle textureHandle)
     {
         arcana::make_task(m_deviceContext->BeforeRenderScheduler(), arcana::cancellation::none(), [this, textureHandle] {
-            if (m_implData->textureBGRA)
+            id<MTLTexture> textureY{};
+            id<MTLTexture> textureCbCr{};
+            @synchronized(m_implData->cameraTextureDelegate) {
+                textureY = [m_implData->cameraTextureDelegate getCameraTextureY];
+                textureCbCr = [m_implData->cameraTextureDelegate getCameraTextureCbCr];
```

Comment:
Do we need to explicitly increment the retain count via CVBufferRetain here or is there an implicit retain/release from taking a reference to this ID?  It looks like we explicitly release the buffer in cleanupTextures.

This blog post seems to suggest that CV Buffers need to be manually ref counted:
https://blog.krybot.com/a?ID=01650-fca75502-f8ee-4678-a905-a50e0de0b1ad

Reply:
The docs at https://developer.apple.com/documentation/corevideo/1456754-cvmetaltexturecachecreatetexture?language=objc say the underlying `CVPixelBuffer` is retained by the texture, so the pixel buffer is retained for as long as we hold a strong reference to the texture. The Xcode iOS AR app template does not explicitly retain the pixel buffer, and I'm not seeing any memory leaks when profiling this change, so I'm fairly certain this is being done correctly here.

Thanks for this. While researching it I found a potential issue in [getCameraTexture](https://github.com/docEdub/BabylonNative/blob/28b4c5c9ac2e28c18f80a083c47f8d6a32da0f59/Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm#L583-L587). In the Xcode iOS AR app template the texture ref is explicitly released on error before returning nil but we're not doing that here or in [XR.mm](https://github.com/docEdub/BabylonNative/blob/28b4c5c9ac2e28c18f80a083c47f8d6a32da0f59/Dependencies/xr/Source/ARKit/XR.mm#L335-L339), which is what this PR is based on. I will update NativeCameraImpl.mm in this PR to plug that potential leak, but changing XR.mm is out of scope for this PR so I'll do another PR for it later if someone else doesn't get to it first.